nixos/lib/eval-config-system.nix

```
@@ -0,0 +1,56 @@
+# DO NOT IMPORT. Use nixpkgsFlake.lib.nixos, or import (nixpkgs + "/nixos/lib")
+{ lib }: # read -^
+
+let
+
+  # Base: Use the miminal module evaluation
+  evalModulesMinimal = (import ./default.nix {
+    inherit lib;
+    # Implicit use of feature is noted in implementation.
+    featureFlags.minimalModules = { };
+  }).evalModules;
+
+  baseModules = import ../modules/module-list.nix;
+
+  # See nixos/doc/manual/development/lib-modules-eval.section.md
+  # or https://nixos.org/manual/nixos/unstable/index.html#sec-modules-evaluation
+  # NOTE: We keep this interface simple, because almost everything can and should be done through `modules`!
+  evalSystemConfiguration = args@{ prefix ? [], modules ? [], specialArgs ? {} }: let
+
+    # Locate Modules: Ensure custom modules are located to guarantee a better user experience.
+    #                 Non-located modules are inline modules and ones imported from value instead
+    #                 of a file reference from which the importer could infer the location.
+    #                 For example when idiomatically consuming a public interface via
+    #                 `flake.nixosModules.my-module` instead of a private one via - say -
+    #                 `(flake + /modules/my-module.nix)`.
+    # Note: `setDefaultModuleLocation` only sets a location, if not set. So if you want to configure
+    #       this yourself, just ensure that all modules in the `modules` list already have one set.
+    modulesLocation = (builtins.unsafeGetAttrPos "modules" args).file or null;
+    locatedModules = if modulesLocation == null then modules
+      else map (lib.setDefaultModuleLocation modulesLocation) modules;
+
+    # `noUserModules` is a contract consumed by
+    # modules/system/activation/specialisation.nix
+    # TODO (during review): critically revise & discuss! (and then remove this line :-)
+    noUserModules = standardModules;
+
+    # Standard Modules: Gather standard modules that are always part of a standard
+    #                   configuration and place them into extra arguments so that
+    #                   a configuration can reproduce a variant of itself inside
+    #                   the module system.
+    standardModules = let
```

Preserving the name for easy comparison. For this PR, making the *change* easy to understand is more important than achieving the perfect end product. Further improvements can be made when the dust settles. (Hoping for no dust at all, but you never know.)

```suggestion
    # A NixOS where almost all of the effects of `modules` are not applied,
    # The `modules` module argument _is_ set to the caller's modules though (is this good??)
    noUserModules = let
```

---

Agreed! We don't need to fix (internal) semantics in this PR, here. Maintaining `noUserModules` but changing the code comment to reflect the new story that there are `baseModules` and `baseModules'` (i.e. `noUserModules`)